setup.cfg

```diff
@@ -38,25 +38,37 @@ package_dir = =src
 packages = find:
 
 [options.extras_require]
+; This is list of minimum dependencies required to run the test.
+; The plan is to remove the `test` extra requirement and skip tests that
+; depend on then
 test =
     cython-test-exception-raiser ~= 1.0
     PyHamcrest >= 1.9.0
-dev =
-    pyflakes >= 1.0.0
-    twisted-dev-tools >= 0.0.2
-    python-subunit
+
+; List of dependencies required to build the documentation and test the
+; release scripts and process.
+dev_release =
     towncrier >= 17.4.0
-    twistedchecker >= 0.7.2
+    pydoctor >= 20.7.0
     # force upgrades for rtd default packages: https://git.io/JU73V
     alabaster~=0.7.12
     commonmark~=0.9.1
     docutils~=0.16.0
-    mock~=4.0
     pillow~=7.2
     readthedocs-sphinx-ext~=2.1
     recommonmark~=0.6.0
     sphinx~=3.2
     sphinx-rtd-theme~=0.5.0
+
+; All the extra tools used to help with the development process.
+dev =
+    %(dev_release)s
+    pyflakes >= 1.0.0
+    twisted-dev-tools >= 0.0.2
+    python-subunit
+    twistedchecker >= 0.7.2
+    mock~=4.0
```

Comment:
mock is for readthedocs

Reply:
Thanks.  I will move it...

But do we need to pin the mock value?

I guess that we only need to pin the high level read the docs and that package will pull its dependencies. Isn't that the case?

Thanks!

Reply:
Thanks. I tried to simplify the dependencies. Since these are dev and release only tools I think that is ok to be more flexible with the dependencies and don't pin everything.


I see that Read The docs build is successful with relaxed dependencies :)